src/components/NotesMindMap/NotesMindMapReducer.js

```diff
@@ -14,114 +14,128 @@ import {
 import { ROOT_NOTE_FOUND_ACTION } from 'components/App/AppActions.js';
 import {
   updateGroupOfOldParent,
-  addGroupTagToNote,
-  removeNoteFromGraph,
+  addGroupTagToNodes,
+  removeNodeFromGraph,
 } from '../../helpers/graph';
 
 const defaultState = {
   selectedNote: {},
   showNoteNameEditor: false,
   noteText: '',
   isChangeParentModeActive: false,
-  notes: [],
+  nodes: [],
   edges: [],
 };
 
 export const notesMindMapReducer = (state = defaultState, { type, data }) => {
   const handleSelectedNoteChildrenFetchedAction = () => {
-    let notes = [...state.notes];
+    let nodes = [...state.nodes];
     let edges = [...state.edges];
 
     const childNotes = data;
     if (childNotes.length) {
       childNotes.forEach(child => {
-        notes.push({
-          id: child.id,
-          label: child.name,
-          name: child.name,
-          isNote: child.isNote,
-          group: 'children',
-        });
+        nodes = addNoteToGraph(nodes, child);
         edges.push({ from: child.parent.id, to: child.id });
       });
-      notes = addGroupTagToNote(notes, state.selectedNote.id);
+      nodes = addGroupTagToNodes(nodes, edges);
     }
-    return { ...state, notes, edges };
+    return {
+      ...state,
+      nodes,
+      edges,
+    };
   };
 
   const handleEditNoteNameAction = () => {
     const noteToEdit = data;
-    return { ...state, showNoteNameEditor: true, selectedNote: noteToEdit };
+    return {
+      ...state,
+      showNoteNameEditor: true,
+      selectedNote: noteToEdit,
+    };
   };
 
   const handleNoteNameUpdateRequestSuccessAction = () => {
     const updatedNote = data;
-    let notes = [...state.notes];
-    notes = notes.filter(note => note.id !== updatedNote.id);
-    notes.push({
-      id: updatedNote.id,
-      label: updatedNote.name,
-      name: updatedNote.name,
-      isNote: updatedNote.isNote,
-    });
-    return { ...state, notes };
+    let nodes = [...state.nodes];
+    nodes = nodes.filter(note => note.id !== updatedNote.id);
+    return {
+      ...state,
+      nodes: addNoteToGraph(nodes, updatedNote),
```

Comment:
We have noDes, then addNoTe... then noDes and then again updatedNoTe.

Looks like a mess. This is an array of noDes, but some elements are noDes and others are noTes... Why?

Reply:
Currently, the functions & actions are using `note` instead of `node`, Do you want me to change everything to `note`?

Reply:
I want no mess in the code :) If `node` and `note` is the same thing, let's rename everything to `node` (I think `node` would be better than `note`, because graphHelper should work with nodes, and the component should render nodes using vis-network).
If `node` and `note` are two different things, then we shouldn't be passing `nodes` to `addNoteToGraph()` function.